packages/host/app/resources/url-bar.ts

```diff
@@ -0,0 +1,106 @@
+import { Resource } from 'ember-resources';
+import { tracked } from '@glimmer/tracking';
+
+interface Args {
+  named: {
+    getValue: () => string | null;
+    setValue?: (val: string) => void;
+    setValueError?: string | null;
+    resetValueError?: () => void;
+  };
+}
+
+export default class URLBarResource extends Resource<Args> {
+  @tracked _value: string | null = null; // url string value
+  @tracked _lastEditedValue: string | null = null; // url string value user when is editing
```

Comment:
If `_` prefixed nomenclature is used to indicate this is a private variable, you can use `private` keyword, `@tracked private lastEditedValue` for example